central/vulnerabilityrequest/manager/requestmgr/manager_impl.go

```diff
@@ -572,3 +630,54 @@ func getReprocessDeploymentMsg(deps ...string) *central.MsgToSensor {
 		},
 	}
 }
+
+func requestName(req *storage.VulnerabilityRequest, lastKnownSeqNumInfo *monthSeqNumPair) string {
+	if req == nil {
+		return ""
+	}
+	if lastKnownSeqNumInfo == nil {
+		lastKnownSeqNumInfo = &monthSeqNumPair{}
+	}
+
+	requestCreatedAt := protoconv.ConvertTimestampToTimeOrNow(req.GetCreatedAt()).UTC()
+	if lastKnownSeqNumInfo.lastCreatedMonth != requestCreatedAt.Month() {
+		lastKnownSeqNumInfo.lastCreatedMonth = requestCreatedAt.Month()
+		lastKnownSeqNumInfo.seqNum = 0
+	}
+
+	lastKnownSeqNumInfo.seqNum = lastKnownSeqNumInfo.seqNum + 1
+
+	userShortName := getShortName(req.GetRequestor())
+	return fmt.Sprintf("%s%s%s%s%d", userShortName, common.VulnReqNameSeparator, requestCreatedAt.Format("060102"), common.VulnReqNameSeparator, lastKnownSeqNumInfo.seqNum)
+}
+
+func getSeqNum(requestName string) int {
+	idx := strings.LastIndex(requestName, common.VulnReqNameSeparator)
+	if idx == -1 {
+		return 0
+	}
+	i, err := strconv.Atoi(requestName[idx+1:])
+	if err != nil {
+		log.Errorf("could not determine the vulnerability request sequence number: %v", err)
+	}
+	return i
+}
+
+func getShortName(user *storage.SlimUser) string {
+	if user == nil {
+		return common.DefaultUserShortName
+	}
+
+	name := strings.ToUpper(user.GetName())
+	parts := strings.Split(name, " ")
+	for i := 0; i < len(parts); i++ {
+		parts[i] = strings.TrimSpace(parts[i])
+	}
+
+	firstName := stringutils.FirstNonEmpty(parts...)
+	lastName := stringutils.LastNonEmpty(parts...)
+	if firstName != "" && lastName != "" {
+		return fmt.Sprintf("%c%c", firstName[0], lastName[0])
```

Comment:
Names are complicated things and there really isn't a good way of getting initials. For someone with only a first name it'll always show SYS (the default). And some folks consider their last name, their first, etc. 

Since it doesn't actually matter that the short name is an initial, why not take the first 2 or 3 chars from the name itself (with spaces removed)?

Reply:
For the folks that have one worded name, the first and the last name would be non-empty; note that we are looking for `LastNonEmpty`. `SYS` will be used only if there is some invalid situation like `" "` API token name.